crates/networking/rpc/clients/eth/mod.rs

```diff
@@ -138,8 +140,35 @@ impl EthClient {
     }
 
     async fn send_request(&self, request: RpcRequest) -> Result<RpcResponse, EthClientError> {
+        // Try first rpc url
+        let mut response = {
+            let url = self
+                .urls
+                .first()
+                .ok_or(EthClientError::Custom("Client has no rpc urls".to_string()))?;
+            self.send_request_to_url(url, &request).await
+        };
+
+        if response.is_ok() {
+            return response;
+        }
+        // If first rpc url fails try with the rest
+        for url in self.urls.iter().skip(1) {
+            response = self.send_request_to_url(url, &request).await;
+            if response.is_ok() {
+                break;
+            }
+        }
+        response
+    }
```

Comment:
Why not go to the for loop directly? Perf optimization?

Reply:
I needed to initialize the variable with something otherwise if none of the rpc's return ok the variable would be uninitialized. I tried to put some thought into refactoring this function to avoid this but couldn't come up with anything better.

Reply:
What about something like?
```suggestion
        for url in self.urls.iter() {
            response = self.send_request_to_url(url, &request).await;
            if response.is_ok() {
                return response;
            }
        }
        return Err(EthClientError::AllRPCFailed)
    }
```